### PR TITLE
update signers for invokefunction and invokescript

### PR DESCRIPTION
**Pull request by ProDog**
Apply https://github.com/neo-project/neo/pull/1752.

Update WitnessScope for signers.

### CODE REVIEW

File: src/RpcServer/RpcServer.SmartContract.cs [hunk 1]
```diff
@@ -74,31 +76,42 @@ private JObject GetInvokeResult(byte[] script, Cosigners cosigners = null)
...
                 json["stack"] = "error: recursive reference";
             }
-            ProcessInvokeWithWallet(json, cosigners);
+            ProcessInvokeWithWallet(json, signers);
             return json;
         }
 
+        private static Signers SignersFromJson(JArray _params)
+        {
+            return new Signers(_params.Select(u => new Signer()
```
Comment (shargon) [inline src/RpcServer/RpcServer.SmartContract.cs:85]: We can serialize it and deserialize it in order to check that all the information it's good

https://github.com/neo-project/neo/blob/54784e03091aa217e133fbe3e70288d36c985e58/src/neo/Network/P2P/Payloads/Signer.cs#L28-L39

Comment (Tommo-L) [reply src/RpcServer/RpcServer.SmartContract.cs:85]: But it's not convenient for the caller.

Comment (shargon) [reply src/RpcServer/RpcServer.SmartContract.cs:85]: If he fill a wrong signer, he's tx will never arrive

Comment (ProDog) [reply src/RpcServer/RpcServer.SmartContract.cs:85]: @shargon can you help modify this😅

**Pull request merged**